### ISSUE TITLE
Consider adding --strict-equality to --strict

### ISSUE THREAD

**Issue by berquist** [python/mypy#7910]
This doesn't report any errors with `--strict-equality`:

```python
def foo() -> bool:
    x: int = 5
    y: str = "foo"
    return x == y


def bar() -> bool:
    x = 5
    y = "bar"
    return x == y


def is_magic(x: bytes) -> bool:
    # Error: Non-overlapping equality check (left operand type: "bytes",
    #        right operand type: "str")  [comparison-overlap]
    return x == "magic"


if __name__ == "__main__":
    print(foo())
    print(bar())
    print(is_magic(b"magic"))
```

even though the third example [is from the docs](https://github.com/python/mypy/blob/d7268ee3336a7d2bba894794a6193872631414a5/docs/source/error_code_list2.rst#check-that-comparisons-are-overlapping-comparison-overlap).

Versions:
- Python: 3.7.4
- mypy: 0.750+dev.59617e8f895ca12a6fb0ec46f378ce384a757118

**Comment by ilevkivskyi** [comment]
I don't know what you are doing, everything works perfectly for me. Maybe you have have a broken install, or something else.

**Comment by berquist** [comment]
I was using `--strict`, not `--strict-equality`.

**Comment by ilevkivskyi** [comment]
> I was using `--strict`, not `--strict-equality`.

Hm, actually now that it is a bit more "mature" we can make it a strict flag, unless @msullivan is against.

See also https://github.com/python/mypy/issues/7469

**Comment by msullivan** [comment]
Fine by me.
